modules/datepicker/basepicker.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react'
+import {Button} from 'react-native'
+
+import {useTheme} from '@frogpond/app-theme'
+import type {AppTheme} from '@frogpond/app-theme'
+
+import moment from 'moment-timezone'
+import type {Moment} from 'moment-timezone'
+
+import DateTimePicker from '@react-native-community/datetimepicker'
+import type {Event} from '@react-native-community/datetimepicker'
+
+import {BaseDatetimePickerProps} from './types'
+
+const FORMATS = {
+	date: 'YYYY-MM-DD',
+	datetime: 'YYYY-MM-DD HH:mm',
+	time: 'HH:mm',
+}
+
+export const BaseDateTimePicker = (
+	props: BaseDatetimePickerProps,
+): JSX.Element => {
+	let [date, setDate] = React.useState(props.initialDate)
+	let [timezone] = React.useState(props.initialDate.tz() || '')
+
+	const onChange = (_event: Event, selectedDate?: Date | undefined) => {
+		// Reset Android's modal's presenting state
+		props.onChange?.()
+
+		if (!selectedDate) {
+			return
+		}
+
+		setDate(moment(selectedDate))
+		props.onDateChange(moment.tz(selectedDate, timezone))
+	}
+
+	const formatDate = (date: Moment): string => {
+		const {mode, format = FORMATS[mode]} = props
+		return moment.tz(date, timezone).format(format)
+	}
+
+	const getTimezoneOffsetInMinutes = (): number => {
+		// We need to negate the offset because moment inverts the offset
+		// for POSIX compatability. So, GMT-5 (CST) is shown to be GMT+5.
+		//
+		// We also need to make the types happy when we negate a negative
+		// MomentZone which is possibly null.
+		let dateInUnixMs = date.valueOf()
+		let momentZone = moment.tz.zone(timezone)
+
+		let offset = momentZone ? -momentZone.utcOffset(dateInUnixMs) : 0
+		return offset
+	}
+
+	let theme: AppTheme = useTheme()
```

Comment:
`useTheme` can return `AppTheme | undefined`. We should handle that second case, or also update `useTheme` to remove that possible return type. (If the latter, we'd also need to update the rest of the places where we might have handling.)

Reply:
I don't think it's expected for us to ever encounter a case that `useTheme` returns `undefined`, since `setTheme` is called inside the initialization path, but TypeScript is telling us here that our code is written in such a way that if `setTheme` _wasn't_ called, then the default theme we specify in `createTheming` (currently, `undefined`, hence the `| undefined` case) would get returned instead.

- In the case of this branch, if you don't want to touch other code, you could just use an `if` to discard the possibility

  ```typescript
  let theme: AppTheme | undefined = useTheme()

  if (!theme) {
    return <></> // or some other sensible default
  }
  ```
  
- Otherwise, couldn't we change out `useTheme` with `getTheme`?

Reply:
Woah, I thought I used `getTheme` but you’re right, I did not!

Reply:
There are a few modules in the app that call `useTheme` that will need what you stated above if they’re kept as-is.